pallets/providers/src/lib.rs

```diff
@@ -511,6 +511,17 @@ impl<T: Config> Pallet<T> {
     }
 }
 
+impl<T: Config> pallet_proofs_dealer::TrieVerifier for Pallet<T> {
+    fn verify_proof(
+        _root: &[u8; 32],
+        _challenges: &[u8; 32],
+        _proof: &pallet_proofs_dealer::CompactProof,
+    ) -> bool {
+        // TODO: implement this
+        todo!()
+    }
+}
+
```

Comment:
I don't think this should be here for two reasons:
- Verifying a proof is a primitive, in my mind this shouldn't be related to the StorageProviders pallet but be its own thing, but I'm not sure if we had agreed on this. cc @ffarall 
- If we decide that StorageProviders will implement the Trie root verification, that shouldn't mean we must introduce a direct dependency to the ProofsDealer pallet, but instead use the support/traits folder and add the trait there